python/triton/language/semantic.py

```diff
@@ -1057,17 +1057,24 @@ def reduce_impl(input: tl.tensor, axis: int, builder: ir.builder, name: str,
         if INT_OP in int_op_to_unit:
             INT_OP = int_op_to_unit[INT_OP]
 
+    # If we are doing an argmin or argmax we want to use an int32 output type
+    out_scalar_ty = scalar_ty
+    if FLOAT_OP is ir.REDUCE_OP.ARGFMAX or INT_OP is ir.REDUCE_OP.ARGMAX:
+        out_scalar_ty = tl.int32
+    elif FLOAT_OP is ir.REDUCE_OP.ARGFMIN or INT_OP is ir.REDUCE_OP.ARGMIN:
+        out_scalar_ty = tl.int32
```

Comment:
Maybe it's better to merge conditions? e.g., 

    if FLOAT_OP in [ir.REDUCE_OP.ARGFMAX, ir.REDUCE_OP.ARGFMIN] or INT_OP in  [ir.REDUCE_OP.ARGMAX, ir.REDUCE_OP.ARGMIN]